command/runner.go

```diff
@@ -203,3 +215,44 @@ func (r *cmdWaitReadCloser) Close() error {
 	r.closed = true
 	return nil
 }
+
+func runPager() (func(), error) {
+	var pager string
+	if p := os.Getenv("KUBEPAGER"); p != "" {
+		pager = p
+	} else if p := os.Getenv("PAGER"); p != "" {
+		pager = p
+	} else {
+		pager = "pager"
+	}
+
+	pargs := strings.Fields(pager)
+	if _, err := exec.LookPath(pargs[0]); err != nil {
+		return nil, err
+	}
+	cmd := exec.Command(pargs[0], pargs[1:]...)
+
+	r, w, err := os.Pipe()
+	if err != nil {
+		return nil, err
+	}
+	cmd.Stdin = r
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	Stdout = colorable.NewColorable(w)
+
+	c := make(chan struct{})
+	go func() {
+		defer close(c)
+		err := cmd.Run()
+		if err != nil {
+			panic(err)
+		}
+		os.Exit(0)
+	}()
+
+	return func() {
+		w.Close()
+		<-c
+	}, nil
```

Comment:
Suggest wrapping the writer and channel in a small struct type that implements `io.Closer`, and then return that struct. This will make the return value's intentions clearer than just returning an anonymous function.

Could also use this return the new stdout, instead of setting that global variable in the function.

Such as:

```go
p, err := runPager()
if err != nil {
    fmt.Printf(...)
}
Stdout = p.Writer()
defer p.Close()